staging/src/k8s.io/apiserver/pkg/admission/plugin/cel/compile.go

```diff
@@ -100,6 +102,8 @@ func buildNoParamsEnv(baseEnv *cel.Env) (*cel.Env, error) {
 	propDecls = append(propDecls, cel.Variable(ObjectVarName, cel.DynType))
 	propDecls = append(propDecls, cel.Variable(OldObjectVarName, cel.DynType))
 	propDecls = append(propDecls, cel.Variable(RequestVarName, requestType.CelType()))
+	propDecls = append(propDecls, cel.Variable(AuthorizerVarName, library.AuthorizerType))
+	propDecls = append(propDecls, cel.Variable(RequestResourceAuthorizerVarName, library.ResourceCheckType))
 
 	opts = append(opts, propDecls...)
 	env, err := baseEnv.Extend(opts...)
```

Comment:
this env is only used by validating admission? just making sure we're not exposing the `authorizer` param further than we intend to

Reply:
This is a great point! This path was planned to be used by webhook match as well. I guess a separation is needed when anyone plan to use this(same suggestion also mentioned in discussion https://github.com/kubernetes/kubernetes/pull/115816#discussion_r1122198963). Since this is not moved into a more shared dir, should we have doc/comment to remind people who is planning to reuse this in the future(especially when we already know one is coming)?

Reply:
I've reorganized the interfaces introduced in https://github.com/kubernetes/kubernetes/pull/115816 to make it possible to declare `authorizer` and binding only in the expressions where we want it.